### PR TITLE
SONARIAC-980 ARM Bicep support: Create memberExpression

### PR DESCRIPTION
**Pull request by jonas-wielage-sonarsource**

### CODE REVIEW

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java [hunk 1]
```diff
@@ -249,6 +250,31 @@ public IfExpression ifExpression(SyntaxToken keyword, ParenthesizedExpression co
...
+    }
+    return result;
+  }
+
+  public MemberExpression memberExpressionComponent(SyntaxToken firstToken, Identifier identifier) {
+    StringLiteralImpl identifierAsStringLiteral = new StringLiteralImpl(((IdentifierImpl) identifier).getToken());
+    return new MemberExpressionImpl(firstToken, identifierAsStringLiteral);
+  }
+
+  public MemberExpression memberExpressionComponent(SyntaxToken firstToken, Expression expression, SyntaxToken secondToken) {
```
Comment (nils-werner-sonarsource) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java:266]: IMO the tokens should be named `openBracket` and `closeBracke` or something similar. Right now it is hard to see from which syntax the method is called.

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/api/bicep/MemberExpression.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+package org.sonar.iac.arm.tree.api.bicep;
+
+import javax.annotation.CheckForNull;
+import org.sonar.iac.arm.tree.api.Expression;
+
+public interface MemberExpression extends Expression {
+  @CheckForNull
+  Expression expression();
+
+  Expression referencingObject();
```
Comment (nils-werner-sonarsource) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/api/bicep/MemberExpression.java:29]: Is this an object? Isn't it more a member access?

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM! I agree with the points raised by Nils, couldn't find more to say :sweat_smile:

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/MemberExpressionImpl.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    this.firstToken = firstToken;
+    this.expression = expression;
+    this.secondToken = null;
+  }
+
+  public MemberExpressionImpl(SyntaxToken firstToken, Expression expression, SyntaxToken secondToken) {
+    this.firstToken = firstToken;
+    this.expression = expression;
+    this.secondToken = secondToken;
+  }
```
[the remark this answers is not in the document]
Comment (jonas-wielage-sonarsource) [reply iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/MemberExpressionImpl.java:59]: But I agree, that would look cleaner, and should be the purpose of a Factory

**Pull request merged**